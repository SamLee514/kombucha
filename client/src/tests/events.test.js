// Test the Events component
import ReactDOM from "react-dom";
import Events from "../views/events/events";
import renderer from "react-test-renderer";

it("renders events.tsx without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Events />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the VUMS logo", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Events />, div);
  expect(div.querySelector(".vumslogo")).toBeTruthy();
});

it("renders the correct title of the page", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Events />, div);
  expect(div.querySelector("h1").textContent).toBe("Events With VUMS");
});

it("loads a default card for testing pruposes", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Events />, div);
  expect(div.querySelector(".ant-card")).toBeTruthy();
});

it("loads the correct testing event title", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Events />, div);
  expect(div.querySelector(".ant-card-meta-title").textContent).toBe(
    "This is a Test Card That Will Be Removed From Production"
  );
});

it("loads the correct testing event description", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Events />, div);
  expect(div.querySelector(".ant-card-meta-description").textContent).toBe(
    "This is a simple test card for the purpose of demonstration for JEST testing in react. It will be removed!"
  );
});
it("matches events page snapshot", () => {
  const tree = renderer.create(<Events />).toJSON();
  expect(tree).toMatchSnapshot();
});
