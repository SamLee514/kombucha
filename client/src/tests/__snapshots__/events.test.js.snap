// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches events page snapshot 1`] = `
<div
  className="centeredCards"
>
  <div
    className="centered"
  >
    <img
      alt="VUMS Logo"
      className="vumslogo"
      src="vums.png"
    />
    <h1>
      Events With VUMS
    </h1>
  </div>
</div>
`;
